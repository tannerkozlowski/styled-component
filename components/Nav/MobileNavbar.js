import React, { PureComponent } from 'react'
import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { navbarHeight } from '../../utils/sizes'
import { paleGrey } from '../../utils/colors'
import { mobile } from '../../utils/media'
import Link from '../Link'
import NavLinks from './NavLinks'
import Social from './Social'
import Search from './Search'
import Logo from './Logo'
import NavSeparator from './NavSeparator'
import NavButton from './NavButton'

const Wrapper = styled.div`
  display: none;

  ${mobile(css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: ${rem(navbarHeight)};
  `)}
`

const SecondaryMenu = styled.div`
  position: absolute;
  top: ${rem(navbarHeight)};
  left: 0;
  right: 0;

  ${p => p.open ? css`
    height: ${rem(navbarHeight)};
  ` : css`
    height: 0;
  `}

  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 ${rem(20)};
  transition: height 0.1s;

  -webkit-user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;

  background: ${paleGrey};
  color: #868686;
`

const LogoLink = styled(Link).attrs({
  unstyled: true,
  href: '/',
})`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`

const ArrowWrapper = styled.div`
  transition: transform 0.1s;

  ${p => p.rotate && css`
    transform-origin: 50% 55%;
    transform: rotate(180deg);
  `}
`

const PaddedSocial = styled(Social)`
  padding-right: ${rem(20)};
`

const CloseIcon = () => (
  <svg xmlns="http://www.w3.org/2000/svg" width="15" height="15" xmlnsXlink="http://www.w3.org/1999/xlink">
    <title>close</title>
    <use fill="#FFF" xlinkHref="#close" transform="translate(1 1)" />
    <defs>
      <path id="close" d="M-.7.7l13 13 1.4-1.4-13-13L-.7.7zm13-1.4l-13 13 1.4 1.4 13-13-1.4-1.4z"/>
    </defs>
  </svg>
)

const FoldIcon = () => (
  <svg xmlns="http://www.w3.org/2000/svg" width="17" height="14" xmlnsXlink="http://www.w3.org/1999/xlink">
    <title>fold</title>
    <use fill="#FFF" xlinkHref="#fold" transform="translate(0 1)"/>
    <defs>
      <path id="fold" d="M0 1h17v-2H0v2zm17 4H0v2h17V5zM0 13h17v-2H0v2z"/>
    </defs>
  </svg>
)

const ArrowIcon = props => (
  <ArrowWrapper {...props}>
    <svg xmlns="http://www.w3.org/2000/svg" width="12" height="8" xmlnsXlink="http://www.w3.org/1999/xlink">
      <title>arrow down</title>
      <use fill="#FFF" xlinkHref="#menuArrow" transform="translate(1 1)"/>
      <defs>
        <path id="menuArrow" d="M5 5l-.7.7.7.7.7-.7L5 5zM9.3-.7l-5 5 1.4 1.4 5-5L9.3-.7zm-3.6 5l-5-5L-.7.7l5 5 1.4-1.4z"/>
      </defs>
    </svg>
  </ArrowWrapper>
)

class MobileNavbar extends PureComponent {
  state = {
    isMenuOpen: false,
  }

  render() {
    const { isMenuOpen } = this.state
    const { isFolded, onFold } = this.props

    return (
      <Wrapper>
        <NavButton
          active={!isFolded}
          onClick={onFold}
        >
          {isFolded ? <FoldIcon /> : <CloseIcon />}
        </NavButton>

        <LogoLink>
          <Logo compact />
        </LogoLink>

        <NavButton
          onClick={this.toggleMenu}
          active={isMenuOpen}
        >
          <ArrowIcon rotate={isMenuOpen} />
        </NavButton>

        <SecondaryMenu open={isMenuOpen}>
          <NavLinks />
          <NavSeparator />
          <Search />
          <NavSeparator />
          <PaddedSocial />
        </SecondaryMenu>
      </Wrapper>
    )
  }

  toggleMenu = () => {
    this.setState({ isMenuOpen: !this.state.isMenuOpen })
  }
}

export default MobileNavbar
