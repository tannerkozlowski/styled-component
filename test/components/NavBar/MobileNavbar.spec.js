import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import MobileNavbar from '../../../components/Nav/MobileNavbar'

test('MobileNavbar renders correctly', () => {
  const tree = renderer.create(<MobileNavbar />).toJSON()

  expect(tree).toMatchSnapshot()
})
