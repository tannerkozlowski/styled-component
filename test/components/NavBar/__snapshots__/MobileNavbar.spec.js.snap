// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavbar renders correctly 1`] = `
.c9 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 0.2777777777777778rem;
  height: 0.2777777777777778rem;
  margin: 0 0.8333333333333334rem;
  border-radius: 50%;
  background: currentColor;
  opacity: 0.35;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  line-height: 2.7777777777777777rem;
  -webkit-transition: opacity 0.2s,-webkit-transform 0.2s;
  -webkit-transition: opacity 0.2s,transform 0.2s;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
  -webkit-letter-spacing: 0.022222222222222223rem color:currentColor;
  -moz-letter-spacing: 0.022222222222222223rem color:currentColor;
  -ms-letter-spacing: 0.022222222222222223rem color:currentColor;
  letter-spacing: 0.022222222222222223rem color:currentColor;
}

.c8:hover,
.c8:focus {
  opacity: 0.8;
}

.c8:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  opacity: 0.6;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c15 {
  display: inline-block;
  margin-right: 1.1111111111111112rem;
  line-height: 2.7777777777777777rem;
  -webkit-transition: opacity 0.2s,-webkit-transform 0.2s;
  -webkit-transition: opacity 0.2s,transform 0.2s;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
}

.c15:last-child {
  margin-right: 0;
}

.c15:hover,
.c15:focus {
  opacity: 0.8;
}

.c15:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  opacity: 0.6;
}

.c15 svg path {
  fill: currentColor;
}

.c16 {
  width: 0.7777777777777778rem;
  height: 0.7777777777777778rem;
}

.c17 {
  width: 1.0555555555555556rem;
  height: 0.8333333333333334rem;
}

.c18 {
  width: 0.9444444444444444rem;
  height: 0.9444444444444444rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c12 {
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 7.222222222222222rem;
  line-height: 1.6666666666666667rem;
  font-size: 0.8333333333333334rem;
  color: currentColor;
}

.c12::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.5;
}

.c12::-moz-placeholder {
  color: currentColor;
  opacity: 0.5;
}

.c12:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
}

.c12::placeholder {
  color: currentColor;
  opacity: 0.5;
}

.c11 {
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2.7777777777777777rem;
  margin-right: 0.2222222222222222rem;
  cursor: pointer;
}

.c11:hover,
.c11:focus {
  opacity: 0.7;
  filter: drop-shadow(0 1px 2px rgba(0,0,0,0.2));
}

.c11:active {
  background: rgba(0,0,0,0.1);
}

.c11 svg {
  width: 0.8888888888888888rem;
  height: 0.9444444444444444rem;
}

.c11 svg path {
  fill: currentColor;
}

.c3 {
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  background-position: center;
  background-size: contain;
  background-image: url(/static/icon.png);
  width: 5.555555555555555rem;
  height: 1.6666666666666667rem;
}

.c1 {
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  min-width: 2.7777777777777777rem;
  height: 2.7777777777777777rem;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
  background: rgba(0,0,0,0.07);
}

.c1:focus {
  border: 0.1111111111111111rem solid currentColor;
  border-radius: 0.1111111111111111rem;
}

.c1 svg {
  display: inline-block;
}

.c1 svg path {
  fill: currentColor;
}

.c4 {
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  min-width: 2.7777777777777777rem;
  height: 2.7777777777777777rem;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
}

.c4:focus {
  border: 0.1111111111111111rem solid currentColor;
  border-radius: 0.1111111111111111rem;
}

.c4 svg {
  display: inline-block;
}

.c4 svg path {
  fill: currentColor;
}

.c0 {
  display: none;
}

.c6 {
  position: absolute;
  top: 2.7777777777777777rem;
  left: 0;
  right: 0;
  height: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  padding: 0 1.1111111111111112rem;
  -webkit-transition: height 0.1s;
  transition: height 0.1s;
  -webkit-user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;
  background: #efefef;
  color: #868686;
}

.c2 {
  display: inline-block;
  vertical-align: center;
  margin-right: 1.9444444444444444rem;
}

.c5 {
  -webkit-transition: -webkit-transform 0.1s;
  -webkit-transition: transform 0.1s;
  transition: transform 0.1s;
}

.c13 {
  padding-right: 1.1111111111111112rem;
}

@media (max-width:62.5em) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: space-between;
    -webkit-justify-content: space-between;
    -ms-flex-pack: space-between;
    justify-content: space-between;
    height: 2.7777777777777777rem;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onClick={undefined}
  >
    <svg
      height="15"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <title>
        close
      </title>
      <use
        fill="#FFF"
        transform="translate(1 1)"
        xlinkHref="#close"
      />
      <defs>
        <path
          d="M-.7.7l13 13 1.4-1.4-13-13L-.7.7zm13-1.4l-13 13 1.4 1.4 13-13-1.4-1.4z"
          id="close"
        />
      </defs>
    </svg>
  </button>
  <a
    className="c2"
    href="/"
    onClick={[Function]}
  >
    <div
      className="c3"
    />
  </a>
  <button
    className="c4"
    onClick={[Function]}
  >
    <div
      className="c5"
      rotate={false}
    >
      <svg
        height="8"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <title>
          arrow down
        </title>
        <use
          fill="#FFF"
          transform="translate(1 1)"
          xlinkHref="#menuArrow"
        />
        <defs>
          <path
            d="M5 5l-.7.7.7.7.7-.7L5 5zM9.3-.7l-5 5 1.4 1.4 5-5L9.3-.7zm-3.6 5l-5-5L-.7.7l5 5 1.4-1.4z"
            id="menuArrow"
          />
        </defs>
      </svg>
    </div>
  </button>
  <div
    className="c6"
    open={false}
  >
    <nav
      className="c7"
    >
      <a
        className="c8"
        href={undefined}
        onClick={[Function]}
      >
        Concepts
      </a>
      <span
        className="c9"
      />
      <a
        className="c8"
        href={undefined}
        onClick={[Function]}
      >
        Try it out
      </a>
      <span
        className="c9"
      />
      <a
        className="c8"
        href={undefined}
        onClick={[Function]}
      >
        Documentation
      </a>
    </nav>
    <span
      className="c9"
    />
    <form
      className="c10"
    >
      <button
        className="c11"
      >
        <svg
          height="16"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <title>
            search
          </title>
          <use
            fill="#FFF"
            xlinkHref="#search-icon"
          />
          <defs>
            <path
              d="M14.772 14.573l-3.698-3.96c.95-1.164 1.472-2.628 1.472-4.153C12.546 2.898 9.732 0 6.273 0 2.813 0 0 2.898 0 6.46s2.814 6.46 6.273 6.46c1.298 0 2.536-.403 3.594-1.17l3.726 3.992c.155.166.365.258.59.258.212 0 .413-.083.566-.235.32-.322.33-.857.02-1.192zm-8.5-12.888c2.558 0 4.637 2.142 4.637 4.775 0 2.633-2.08 4.775-4.64 4.775-2.56 0-4.64-2.142-4.64-4.775 0-2.633 2.08-4.775 4.637-4.775z"
              id="search-icon"
            />
          </defs>
        </svg>
      </button>
      <input
        className="c12"
        placeholder="Search ..."
      />
    </form>
    <span
      className="c9"
    />
    <nav
      className="c13 c14"
    >
      <a
        className="c15"
        href="https://spectrum.chat/styled-components/"
        onClick={[Function]}
      >
        <svg
          className="c16"
          height="14"
          viewBox="0 0 15 15"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <title>
            spectrum-logo
          </title>
          <use
            fill="#FFF"
            xlinkHref="#spectrum"
          />
          <defs>
            <path
              d="M0 6.5V1c0-.6.4-1 1-1 9 .3 13.7 5 14 14 0 .6-.4 1-1 1H8.5c-.6 0-1-.4-1-1-.3-4.4-2-6.2-6.5-6.5-.6 0-1-.4-1-1z"
              id="spectrum"
            />
          </defs>
        </svg>
      </a>
      <a
        className="c15"
        href="https://twitter.com/someone"
        onClick={[Function]}
      >
        <svg
          className="c17"
          height="15"
          viewBox="0 0 19 15"
          width="19"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <title>
            twitter-logo
          </title>
          <use
            fill="#FFF"
            xlinkHref="#b"
          />
          <defs>
            <path
              d="M18.2 1.8l-2 .6c.6-.5 1.2-1.2 1.5-2l-2.4.8C14.7.5 13.7 0 12.6 0 10.6 0 9 1.7 9 3.8v1C6 4.4 3 3 1.3.8 1 1 .8 1.8.8 2.4c0 1.3.6 2.5 1.6 3-.6 0-1.2 0-1.7-.3 0 2 1.3 3.7 3 4H2c.5 1.6 2 2.7 3.5 2.7-1.2 1-3 1.6-4.6 1.6H0c1.7 1 3.6 1.7 5.7 1.7 7 0 10.7-6 10.7-11v-.5c.7-.5 1.3-1.2 1.8-2z"
              id="b"
            />
          </defs>
        </svg>
      </a>
      <a
        className="c15"
        href="https://github.com/styled-components"
        onClick={[Function]}
      >
        <svg
          className="c18"
          height="17"
          viewBox="0 0 17 17"
          width="17"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <title>
            github-logo
          </title>
          <use
            fill="#FFF"
            xlinkHref="#a"
          />
          <defs>
            <path
              d="M15.86 4.34c-.76-1.335-1.79-2.392-3.093-3.17C11.463.39 10.042 0 8.5 0 6.958 0 5.536.39 4.233 1.17 2.93 1.947 1.9 3.004 1.14 4.34.38 5.674 0 7.132 0 8.713c0 1.898.54 3.606 1.62 5.122 1.082 1.517 2.48 2.566 4.19 3.148.2.038.347.01.443-.08.096-.09.144-.203.144-.34l-.006-.612v-1l-.25.05c-.16.03-.36.05-.61.04-.24 0-.5-.03-.77-.08-.26-.05-.51-.16-.74-.34-.23-.18-.39-.41-.49-.7l-.11-.26c-.07-.17-.19-.36-.35-.58-.16-.21-.32-.35-.48-.43l-.08-.05c-.05-.04-.09-.08-.14-.132-.04-.053-.08-.106-.1-.16-.02-.053 0-.097.06-.13.06-.035.17-.052.32-.052l.22.036c.15.03.33.12.55.27.22.154.4.35.54.59.17.31.375.55.615.71.24.164.484.245.727.245s.456-.02.63-.056c.18-.037.346-.094.5-.17.067-.507.248-.896.54-1.17-.42-.043-.795-.112-1.13-.203-.34-.09-.685-.24-1.04-.44-.36-.2-.657-.455-.894-.76-.234-.3-.43-.7-.58-1.19-.15-.49-.225-1.06-.225-1.7 0-.917.29-1.695.875-2.34-.27-.69-.244-1.46.08-2.313.214-.07.53-.015.95.155.42.17.73.314.926.435.195.12.35.225.47.308.684-.2 1.394-.3 2.124-.3s1.44.1 2.124.297l.42-.27c.288-.184.628-.35 1.02-.5.39-.15.69-.195.895-.126.334.857.364 1.63.09 2.317.584.645.875 1.42.875 2.34 0 .64-.072 1.21-.223 1.707-.15.5-.343.897-.583 1.196-.24.298-.54.55-.896.754-.355.202-.702.35-1.04.44-.334.09-.712.16-1.132.205.38.34.57.875.57 1.61v2.39c0 .136.045.25.137.34.096.09.24.118.44.08 1.71-.582 3.11-1.632 4.19-3.15 1.06-1.525 1.6-3.235 1.6-5.13 0-1.58-.38-3.04-1.14-4.37z"
              id="a"
            />
          </defs>
        </svg>
      </a>
    </nav>
  </div>
</div>
`;
